appcli/configuration_manager.py

```diff
@@ -215,19 +220,40 @@ def migrate_configuration(
                 "Migrated variables did not pass application-specific variables validation function."
             )
 
-        overrides_exist = self.__check_overrides_exist()
-        override_dir: Path
-        if overrides_exist:
-            override_dir = self.__backup_overrides_directory()
+        baseline_template_overrides_dir = (
+            self.cli_context.get_baseline_template_overrides_dir()
+        )
+        override_backup_dir: Path = self.__backup_directory(
+            baseline_template_overrides_dir
+        )
+
+        configurable_templates_dir = self.cli_context.get_configurable_templates_dir()
+        configurable_templates_backup_dir = self.__backup_directory(
+            configurable_templates_dir
+        )
 
         # Backup and remove the existing generated config dir since it's now out of date
         self.__backup_and_create_new_generated_config_dir(config_version)
 
         # Initialise the new configuration branch and directory with all new files
         self.__create_new_configuration_branch_and_files(config_repo)
 
-        if overrides_exist:
-            self.__copy_in_overrides(override_dir)
+        self.__copy_directory_contents_with_overwrite(
+            override_backup_dir, baseline_template_overrides_dir
+        )
+        if self.__directory_is_not_empty(baseline_template_overrides_dir):
+            logger.warn(
+                f"Overrides directory [{baseline_template_overrides_dir}] is non-empty, please check for compatibility of overridden files"
+            )
+
+        # We need to overwrite copy so that the configurable templates doesn't include the set of default configurable templates
```

Comment:
An overwrite may not be enough. I think you need to delete the default configurable templates dir.
For example, the default configurable templates dir may contain a file called `example.yml`, which we have deleted on site (since it contains example pages or something). We don't want this to re-appear on an upgrade

Reply:
Careful about naming function